### ISSUE TITLE
Downloading a Production Line with spoiling recipe attempts to index a null factory

### ISSUE THREAD

**Issue by pedter** [Helfima/helmod#567]
Using the "Download Production Line" button throws an error when it attempts to import a recipe without a factory, such as spoilage of Iron Bacteria -> Iron Ore:

![Untitled](https://github.com/user-attachments/assets/2e7e8e85-a601-4972-b5dd-ffcd37c8e21f)

Attempting to download such a production line results in the following error:
```
---- Error ----
__helmod__/data/ModelBuilder.lua:721: attempt to index local 'factory' (a nil value)
---- Feature Flags ----
expansion_shaders:true
freezing:true
quality:true
rail_bridges:true
segmented_units:true
space_travel:true
spoiling:true
---- Mods ----
base:2.0.15
elevated-rails:2.0.15
helmod:2.0.6
quality:2.0.15
space-age:2.0.15
```
Production line download:
```
eNqNU01v5CAM/SujnFlpspceKi6972WuqxUi4GnZEoz4SDeK8t9rQmbKTPfrFOPYD7/3jMaDRSXtQfBFWRkj776hBtsxo3k3llA89B3DNweBdx50gtCxgZpeBZX0exgQ0xXhqaQqQv1LAMZp+MWPzMkReGcCui+DVARm5Cd0hdklgvb4Rukj9T4H0AZcinz5ftf8gy+/xUyzL8kEY8fkuCPGJBPwr+vKfECdVQOIAe6wSuZvMD3BqBdjdQDHl1N/5X8CZTxUAU7/4l4viB6NJZ4N+TqgQVdEBqnQ0bArG2QEkQwhFYGszaVE7KMdGZCSz7NImKSlozdTMaZlccdwrTcKa0ZT7q0nDeAbB7bvXnI5bSXHjxnauCm9ZGp5Ub5tkKRdEre71g67E32RURjnc+JnaSMwHH4CaTPBJ/9eYW7tm6TNu1M3LP+H0B9JsJjHUYaZL0Mm+8k2yl3Ca9c1sTedyXIMBqqLH4YS8Wy37Hoj/lqf1lZz8wJWFiBiDqpC7RJNEGJdlrpufRGXrKY1ozI4n2Hb9ugBdOG0r9dk0kzHMkwe/b2Pm2FhW+ftta+PAVIO7iAewel3L3KNfQ==
```

**Comment by Helfima** [comment]
fixed in next version v2.0.7

**Comment by pedter** [comment]
Finally got around to testing (using 2.0.11) - all looks functional, including downloading old production line strings. Thanks!
